RecoLocalMuon/GEMRecHit/src/ME0RecHitProducer.cc

```diff
@@ -0,0 +1,104 @@
+/** \file
+ *
+ *  $Date: 2013/04/24 17:16:35 $
+ *  $Revision: 1.1 $
+ *  \author M. Maggi -- INFN Bari
+*/
+
+#include "ME0RecHitProducer.h"
+
+
+#include "FWCore/Framework/interface/Event.h"
+#include "FWCore/ParameterSet/interface/ParameterSet.h"
+#include "FWCore/Framework/interface/ESHandle.h"
+
+#include "DataFormats/GEMDigi/interface/ME0DigiPreRecoCollection.h"
+
+#include "Geometry/GEMGeometry/interface/ME0EtaPartition.h"
+#include "Geometry/GEMGeometry/interface/ME0Geometry.h"
+#include "Geometry/Records/interface/MuonGeometryRecord.h"
+#include "DataFormats/MuonDetId/interface/ME0DetId.h"
+#include "DataFormats/GEMRecHit/interface/ME0RecHit.h"
+
+#include "RecoLocalMuon/GEMRecHit/interface/ME0RecHitBaseAlgo.h"
+#include "RecoLocalMuon/GEMRecHit/interface/ME0RecHitAlgoFactory.h"
+#include "DataFormats/GEMRecHit/interface/ME0RecHitCollection.h"
+
+#include <string>
+
+
+using namespace edm;
+using namespace std;
+
+
+ME0RecHitProducer::ME0RecHitProducer(const ParameterSet& config){
+
+  produces<ME0RecHitCollection>();
+  theME0DigiLabel = config.getParameter<InputTag>("me0DigiLabel");
+  
+  // Get the concrete reconstruction algo from the factory
+
+  string theAlgoName = config.getParameter<string>("recAlgo");
+  theAlgo = ME0RecHitAlgoFactory::get()->create(theAlgoName,
+						config.getParameter<ParameterSet>("recAlgoConfig"));
+}
+
+
+ME0RecHitProducer::~ME0RecHitProducer(){
+  delete theAlgo;
+}
+
+
+
+void ME0RecHitProducer::beginRun(const edm::Run& r, const edm::EventSetup& setup){
+}
+
+
+
+void ME0RecHitProducer::produce(Event& event, const EventSetup& setup) {
+
+  // Get the ME0 Geometry
+  ESHandle<ME0Geometry> me0Geom;
+  setup.get<MuonGeometryRecord>().get(me0Geom);
+
+  // Get the digis from the event
+
+  Handle<ME0DigiPreRecoCollection> digis; 
+  event.getByLabel(theME0DigiLabel,digis);
```

Comment:
all code has to be "modernized" for 75X releases:
- use stream or global producer base class https://indico.cern.ch/event/351556/contribution/2/material/slides/0.pdf  https://twiki.cern.ch/twiki/bin/view/CMSPublic/FWMultithreadedFrameworkModuleTypes#Stream_Module
- declare all consumed products and use getByToken https://twiki.cern.ch/twiki/bin/view/CMSPublic/SWGuideEDMGetDataFromEvent#Registering_for_data_access

These are prerequisites to get to 75X